### PR TITLE
Adding support for SqlPoolDeleteRestorePoints and RestorableDroppedSqlPools apis

### PR DESCRIPTION
**Pull request by tunafish88**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [X] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (akning-ms) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3992]: suppression for Lint R4011 for missing response code 204 for a delete operation, need ARM approval. pls provide your justification here for ARM team review

**Comment by tunafish88** [comment]
> ### [Staging] Swagger Validation Report
> `` ️`️✔️` ``BreakingChange [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11257/checks?check_run_id=1271669795)]
> `` ️`❌` ``LintDiff: 1 Errors, 1 Warnings [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11257/checks?check_run_id=1271669801)] [Expand]
> `` ️`️✔️` ``Avocado [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11257/checks?check_run_id=1271669806)]
> `` ️`️✔️` ``ModelValidation [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11257/checks?check_run_id=1271669809)]
> `` ️`️✔️` ``SemanticValidation [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11257/checks?check_run_id=1271669814)]
> Posted by Swagger Pipeline | [How to fix these errors?](https://aka.ms/ci-fix)

- SQL Pools APIs are proxy APIs that call SQL DB APIs. The SQL DB delete restore points API only supports return method 200, so we cannot support 204. It is not possible for the SQL DB team to add 204 support for delete restore points, hence asking for a suppression exception at this time

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3837]: You definitely don't need to (and shouldn't) list async operationResults in your SDK or API.  You typically don't need to expose operationResults at all in the SDK since the SDKs will automatically follow operationResults links from async operations.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3987]: If you are purely proxying these through then you know what restore points the user has created via your APIs and can return 204 appropriately correct?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3837]: Ah, I see this has nothing to do with async operations.  It should be renamed so it isn't using the same resource type name as async operation results (https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#202-accepted-and-location-headers)

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3987]: Are you suggesting we detect the present of content, and return 204 is content is absent, and 200 otherwise ?

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3837]: We've based this change off of swagger documentation for this:

https://docs.microsoft.com/en-us/rest/api/sql/transparentdataencryptionactivities/listbyconfiguration

"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Synapse/workspaces/{workspaceName}/sqlPools/{sqlPoolName}/transparentDataEncryption/{transparentDataEncryptionName}/activities? 

Does the above look ok ?

Comment (pilor) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3837]: Yes, thank you

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3987]: Yes.  A 200 should only be returned if something was actually deleted.  SQL has been around forever and has a bunch of bad, old legacy behavior in their APIs.  We shouldn't carry that forward

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (idear1203) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3902]: >RestorePoints_Get [](start = 24, length = 17)

SqlPoolRestorePoints_Get

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (idear1203) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3957]: >RestorePoints_Delete [](start = 24, length = 20)

SqlPoolRestorePoints_Delete

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (idear1203) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3842]: >TransparentDataEncryptionActivities_ListByConfiguration [](start = 24, length = 55)

SqlPoolTransparentDataEncryptionActivities_ListByConfiguration

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3843]: I'll update this.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3898]: Will change

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3928]: For legacy reasons we will stay this way for now.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3933]: Provided justification down below. the ARM reviewer is aware that we're calling an old sql API behind the scenes which does not have 204 response, so we cannot support it here.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json]: As a result of some confusion in the Synapse RP side, with this Transparent dataencryption activities /operationResults API, we've decided to hold off on this change. So resolving this comment as this code is not going to be there anymore in the PR

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json]: Not supporting this for now. I will send and email to the concerned parties and include you in it.

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json
Comment (pilor) [inline specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json:587]: How do you restore these? Also, typically these soft-delete scenarios give users the ability to purge the restorable objects by calling a DELETE API on the "restorable" variant.

**Review by pilor — CHANGES_REQUESTED** [review]

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json
Comment (pilor) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json:3933]: Yes, unfortunately we'll have to allow it in this scenario

File: specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json
Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json:587]: SO there are 2 options:
('les common)
1. user calls this API (visa some rest client, or via commandlet) and reaches out to CSS who creates an ICM  and we can help the user out

2. There is going to be an Azure portal blade which will call this API to show a drop down of the restorabled droppedpools. The user chooses one of them and the chosen one gets passed onto a Create from 'restorable dropped sql pool' flow., to create a new sql pool.

Comment (pilor) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json:587]: What API is used in #2?  How do you specify via the API that you want to restore one of these databases?  Where is that API change?

Comment (tunafish88) [reply specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/workspace.json:587]: It would have to be the 'create sqlpool' API.  ( sqlpool.json file) with some update to it (to take some more parameters as per my understanding).

This line below if from a Colleague:
'o	Synapse RP would need to expose additional parameters (deletionTime, deletionDateString, deletionTimeString, deletedDatabaseId) in their create sqlpool api (to support the provided arm template)'

**Review by pilor — APPROVED** [review]
Approving as a read-only view of restorable pools that can be restored via a support ticket.  Please open another review for the API that allows restoring these when you implement that.

**Pull request merged**
